src/renku_crac/config.py

```diff
@@ -31,11 +47,47 @@ def from_env(cls):
         """Create a config from environment variables."""
 
         prefix = ""
-        async_sqlalchemy_url = os.environ.get(f"{prefix}ASYNC_SQLALCHEMY_URL", "sqlite+aiosqlite:///data_services.db")
-        sync_sqlalchemy_url = os.environ.get(f"{prefix}SYNC_SQLALCHEMY_URL", "sqlite:///data_services.db")
+        user_store: models.UserStore
+        authenticator: models.Authenticator
         version = os.environ.get(f"{prefix}VERSION", "0.0.1")
+        keycloak_url = None
+        keycloak_realm = "Renku"
+
+        if os.environ.get(f"{prefix}DUMMY_STORES", "false").lower() == "true":
+            async_sqlalchemy_url = os.environ.get(
+                f"{prefix}ASYNC_SQLALCHEMY_URL", "sqlite+aiosqlite:///data_services.db"
+            )
+            sync_sqlalchemy_url = os.environ.get(f"{prefix}SYNC_SQLALCHEMY_URL", "sqlite:///data_services.db")
+            authenticator = DummyAuthenticator(admin=True)
+            user_always_exists = os.environ.get("DUMMY_USERSTORE_USER_ALWAYS_EXISTS", "true").lower() == "true"
+            user_store = DummyUserStore(user_always_exists=user_always_exists)
+        else:
+            async_sqlalchemy_url = os.environ.get(f"{prefix}ASYNC_SQLALCHEMY_URL", "")
+            sync_sqlalchemy_url = os.environ.get(f"{prefix}SYNC_SQLALCHEMY_URL", "")
+            if async_sqlalchemy_url == "" or sync_sqlalchemy_url == "":
+                raise errors.ConfigurationError(message="The sqlalchemy url has to be specified.")
+            keycloak_url = os.environ.get(f"{prefix}KEYCLOAK_URL")
+            if keycloak_url is None:
+                raise errors.ConfigurationError(message="The Keycloak URL has to be specified.")
+            keycloak_realm = os.environ.get(f"{prefix}KEYCLOAK_REALM", "Renku")
+            oidc_disc_data = _oidc_discovery(keycloak_url, keycloak_realm)
+            jwks_url = oidc_disc_data.get("jwks_uri")
+            if jwks_url is None:
+                raise errors.ConfigurationError(
+                    message="The JWKS url for Keycloak cannot be found from the OIDC discovery endpoint."
+                )
+            algorithms = os.environ.get(f"{prefix}KEYCLOAK_TOKEN_SIGNATURE_ALGS")
+            if algorithms is None:
+                raise errors.ConfigurationError(message="At least one token signature algorithm is requried.")
+            algorithms_lst = [i.strip() for i in algorithms.split(",")]
+            jwks = PyJWKClient(jwks_url)
+            authenticator = KeycloakAuthenticator(jwks=jwks, algorithms=algorithms_lst)
+            user_store = KcUserStore(keycloak_url=keycloak_url, realm=keycloak_realm)
+
```

Comment:
Should we use `dataconf` to make it easier to do this kind of thing instead of all the manual `os.environ` calls?

Reply:
Ralf I would like to address this in a separate PR if that is ok wiht you?

Reply:
yes that's perfectly fine